### PR TITLE
Optimize WeightedSumOp for two inputs

### PR DESCRIPTION
**Pull request by xiaomengy**
Summary: Optimize WeightedSumOp for four inputs (2 weighted inputs), just call one kernel instead of two kernels.

Differential Revision: D9566692

### CODE REVIEW

**Comment by BIT-silence** [comment]
@pytrochbot retest this please

**Comment by BIT-silence** [comment]
Some benchmark on GPU

input_shape = [100, 100, 100, 500],
param = model.WeightedSum([param, One, gradient, lr], param)
orginal: 46.43ms
new CUDA: 32.15ms
new CUDNN: 31.16ms

So this diff can reduce about 30% time of WeightedSumOp on training jobs.

File: caffe2/operators/utility_ops.h [hunk 1]
```diff
@@ -321,45 +321,70 @@ class WeightedSumOp : public Operator<Context> {
...
+        X1.template data<T>(),
         weight0.template data<float>(),
-        X0.template data<DstType>(),
-        output->template mutable_data<DstType>(),
+        Y_data,
         &context_);
-    for (int i = 2; i < InputSize(); i += 2) {
-      auto& X = Input(i);
+    for (int i = 4; i < input_size; i += 2) {
+      const auto& Xi = Input(i);
```
Comment (houseroad) [inline caffe2/operators/utility_ops.h]: good name :-)

**Review by houseroad — APPROVED** [review]
LGTM

**Pull request closed without merging**